File: app/src/main/java/com/example/foolcardgame/domain/engine/Deck.kt
package com.example.foolcardgame.domain.engine

import com.example.foolcardgame.domain.model.Card
import com.example.foolcardgame.domain.model.Rank
import com.example.foolcardgame.domain.model.Suit
import kotlin.random.Random

object Deck {

    const val SIZE = 36

    fun create36(): List<Card> = buildList {
        for (suit in Suit.entries) {
            for (rank in Rank.entries) {
                add(Card(suit, rank))
            }
        }
    }

    /** Детерминированное тасование для тестов и воспроизводимых офлайн-игр. */
    fun shuffled(seed: Long): List<Card> {
        val cards = create36().toMutableList()
        cards.shuffle(Random(seed))
        return cards
    }
}

File: server/src/internal/engine/deck.go
package engine

import (
	"math/rand"
)

const deckSize = 36

// Create36 — полная колода 36 карт.
func Create36() []Card {
	out := make([]Card, 0, deckSize)
	for s := SuitSpades; s <= SuitClubs; s++ {
		for r := RankSix; r <= RankAce; r++ {
			out = append(out, Card{Suit: s, Rank: r})
		}
	}
	return out
}

// Shuffled — детерминированное тасование (seed для тестов и воспроизводимых партий).
func Shuffled(seed int64) []Card {
	cards := Create36()
	r := rand.New(rand.NewSource(seed))
	r.Shuffle(len(cards), func(i, j int) {
		cards[i], cards[j] = cards[j], cards[i]
	})
	return cards
}
